### ISSUE TITLE
Propagate error messages from p2pd even if it does not terminate

### ISSUE THREAD

**Issue by justheuristic** [learning-at-home/hivemind#367]
We sometimes get the following error:
```
Traceback (most recent call last):
  File "run_trainer.py", line 203, in <module>
    main()
  File "run_trainer.py", line 148, in main
    dht = hivemind.DHT(
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/dht/__init__.py", line 98, in __init__
    self.run_in_background(await_ready=await_ready)
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/dht/__init__.py", line 143, in run_in_background
    self.wait_until_ready(timeout)
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/dht/__init__.py", line 146, in wait_until_ready
    self._ready.result(timeout=timeout)
  File "/slot/sandbox/nv_tmpfs/d/in/script/0_script_unpacked/hivemind/hivemind/utils/mpfuture.py", line 251, in result
    return super().result(timeout)
  File "/anaconda3/lib/python3.8/concurrent/futures/_base.py", line 439, in result
    return self.__get_result()
  File "/anaconda3/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
hivemind.p2p.p2p_daemon.P2PDaemonError: Daemon failed to start in 15.0 seconds
```

It would be nice to know what killed it.

**Comment by borzunov** [comment]
__TODO:__

- [ ] Clean debug output by removing "MPFuture is destroyed" and "Called NoOpScheduler.step" messages (they become 99% of the output)

**Comment by borzunov** [comment]
Resolved in #375. Some messages are hidden by default, use `LOGLEVEL=DEBUG` to show them.
